Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,16 +1,58 @@
-function setAlarm() {}
+var input_time = document.querySelector("#alarmSet");
+var title_time = document.querySelector("#titleTime");
+var timeRemaining = document.querySelector("#timeRemaining");
+var bUserStopped = false;
+function setAlarm() {
+  
+  let totalSeconds = parseInt(input_time.value, 10);
+  if (isNaN(totalSeconds) || totalSeconds <= 0) {
+    alert("Please enter a valid number of seconds.");
+    return;
+  }
+  
+  const title_second = input_time.value % 60;
+  const title_minute = (input_time.value - title_second) / 60;
+
+  title_time.innerText = `${title_minute.toString().padStart(2, "0")}:${title_second.toString().padStart(2, "0")}`;
+
+  let timer = setInterval(() =>{
+    let seconds = totalSeconds % 60;
+    let minutes = (totalSeconds - seconds) / 60;
+    timeRemaining.innerText = `Time Remaining: ${minutes.toString().padStart(2, "0")}:${seconds.toString().padStart(2, "0")}`;
+    
+    if(totalSeconds <= 0){
+      clearInterval(timer);
+      playAlarm();
+      document.getElementById("set").disabled = false;
+    }
+    else if(bUserStopped){
+      clearInterval(timer);
+      document.getElementById("set").disabled = false;
+
+    }
+    else{
+      document.getElementById("set").disabled = true;
```

Comment:
Interestingly, this only disables the button one second after the timer starts running (and only updates the displayed time 1 second after pressing the button) - I think most users would expect the time displayed to _instantly_ change when they press the button, and for the button to become disabled instantly too.

Can you work out why there's the delay and fix it?

Reply:
tried to fix it and looks ok for me now. but maybe I'm wrong , can you please check again?
thanks